src/Raven.Client/Documents/BulkInsert/BulkInsertOperation.cs

```diff
@@ -348,63 +399,78 @@ public async Task StoreAsync(object entity, string id, IMetadataDictionary metad
         {
             using (ConcurrencyCheck())
             {
-                VerifyValidId(id);
-
-                await ExecuteBeforeStore().ConfigureAwait(false);
-                
-                if (metadata == null)
-                    metadata = new MetadataAsDictionary();
-
-                if (metadata.ContainsKey(Constants.Documents.Metadata.Collection) == false)
+                var strLock = await _streamLock.ReaderLockAsync().ConfigureAwait(false);
```

Comment:
Let's add this to the `ConcurrencyCheck`

Reply:
done